orm/codec.proto

```diff
@@ -11,3 +13,11 @@ message MultiRef {
 message Counter {
   int64 count = 1;
 }
+
+// VersionedID is the combination of document ID and version number.
+message VersionedIDRef {
+  // Unique identifier
+  bytes id = 4 [(gogoproto.customname) = "ID"];
+  // Document version, starting with 1.
+  uint32 version = 5;
```

Comment:
uint32 compresses the data in such a way that the length is vairable and byte comparison of encoded value witll show 100 > 200.

I would have recommended fixed32 https://developers.google.com/protocol-buffers/docs/proto3#scalar But then I read:

Non-varint numeric types are simple – double and fixed64 have wire type 1, which tells the parser to expect a fixed 64-bit lump of data; similarly float and fixed32 have wire type 5, which tells it to expect 32 bits. In both cases the values are stored in **little-endian byte order**. 

https://developers.google.com/protocol-buffers/docs/encoding

So... maybe we don't protobuf encode this one

Reply:
If we don't encode this via protobuf it may become hard for a client to access the data as they would need to match our custom encoding format. 🤔